### PR TITLE
[ACR] Fix some bugs on uncovered cmds

### PR DESCRIPTION
**Pull request by 00Kai0**
**Description**

Fix some bugs on uncovered  acr cmds

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ACR

**Review by rosanch — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acr/build.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def acr_build(cmd,  # pylint: disable=too-many-locals
 
     platform_os, platform_arch, platform_variant = get_validate_platform(cmd, platform)
 
-    DockerBuildRequest, PlatformProperties = cmd.get_models('DockerBuildRequest', 'PlatformProperties')
+    DockerBuildRequest, PlatformProperties = cmd.get_models('DockerBuildRequest', 'PlatformProperties', operation_group='runs')
```
Comment (northtyphoon) [inline src/azure-cli/azure/cli/command_modules/acr/build.py:100]: How do you determine the operation group name? Is there a list that track what modules included in each group?

File: src/azure-cli/azure/cli/command_modules/acr/connected_registry.py [hunk 2]
```diff
@@ -129,7 +129,7 @@ def acr_connected_registry_create(cmd,  # pylint: disable=too-many-locals, too-m
     )
 
     try:
-        return client.create(subscription_id=subscription_id,
+        return client.begin_create(subscription_id=subscription_id,
```
Comment (northtyphoon) [inline src/azure-cli/azure/cli/command_modules/acr/connected_registry.py:132]: why do you need to change the code from xyz to begin_xyz? What is the difference between the two?

File: src/azure-cli/azure/cli/command_modules/acr/connected_registry.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def acr_connected_registry_create(cmd,  # pylint: disable=too-many-locals, too-m
                        "Enabling the data endpoint might affect your firewall rules.\nTo enable data endpoint run:" +
                        "\n\taz acr update -n {} --data-endpoint-enabled true".format(registry_name))
 
-    ErrorResponseException = cmd.get_models('ErrorResponseException')
+    from azure.core.exceptions import HttpResponseError as ErrorResponseException
```
Comment (northtyphoon) [inline src/azure-cli/azure/cli/command_modules/acr/connected_registry.py:79]: Is the type same as before?

File: src/azure-cli/azure/cli/command_modules/acr/build.py (hunk 1 above)
Comment (00Kai0) [reply src/azure-cli/azure/cli/command_modules/acr/build.py:100]: It's defined in [_shared.py](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/profiles/_shared.py)

Comment (rosanch) [reply src/azure-cli/azure/cli/command_modules/acr/build.py:100]: check the file src\azure-cli-core\azure\cli\core\profiles\_shared.py
Pretty much the default version (when not specified) is 2020-11-01-preview'
and all the other defined groups in the file are referring the same API version:
'agent_pools': '2019-06-01-preview',
'tasks': '2019-06-01-preview',
'task_runs': '2019-06-01-preview',
'runs': '2019-06-01-preview',

File: src/azure-cli/azure/cli/command_modules/acr/connected_registry.py (hunk 2 above)
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/acr/connected_registry.py:132]: https://github.com/Azure/azure-cli/blob/dev/doc/track_2_migration_guidance.md#long-running-operation-function-name-change, no behavior change, only name change.

File: src/azure-cli/azure/cli/command_modules/acr/connected_registry.py (hunk 3 above)
Comment (00Kai0) [reply src/azure-cli/azure/cli/command_modules/acr/connected_registry.py:79]: Basically, yes. Track2 SDK has removed all these Exceptions in models.

File: src/azure-cli/azure/cli/command_modules/acr/connected_registry.py (hunk 2 above)
Comment (00Kai0) [reply src/azure-cli/azure/cli/command_modules/acr/connected_registry.py:132]: Refer to this [Guide](https://devblogs.microsoft.com/azure-sdk/migrating-python-management-libraries/#long-running-operations-lro-method-names)

**Review by northtyphoon — APPROVED** [review]
LGTM

**Review by jsntcy — APPROVED** [review]

**Comment by ndrmnl** [comment]
Hi, could it be that this PR introduced the bug in #18841? The error message points out to an error at line 106 here https://github.com/Azure/azure-cli/commit/a8abfbbffe6b3f5d45f6821dabe060871bd1db25?branch=a8abfbbffe6b3f5d45f6821dabe060871bd1db25&diff=unified#diff-0f3568ab838e8abfa5fd9ff9c4e886aa58b3e0d969968a166268ef29e51c6733L100, supposedly `DockerBuildRequest` is returned as a `None` for some reason...

Apologies if it isn't related, seems like a good starting point to investigate the issue.

**Comment by daniel347x** [comment]
This is a show-stopper issue for me. I depend on a running container and at this time I cannot build the container with changes.

**Pull request merged**